File: ruby/Prototype/prototype_pattern.rb
require_relative 'car'
require_relative 'prototype_factory'

baseCar = Car.new
baseCar.name = 'Focus S'
baseCar.printDetails

print "\n"
prototypeFactory = PrototypeFactory.new
otherVersion  = prototypeFactory.____________; # << 1
otherVersion.name = 'Focus SE'
otherVersion.automaticClimateControl = true
otherVersion.cruiseControl = true
otherVersion.printDetails
           

File: java/Prototype/src/PrototypePattern.java
package exercise;
public class PrototypePattern
{
    public static void main(String[] args)
    {
        try
        {
        	Car baseCar = new Car();
        	baseCar.setName("Focus S");
        	baseCar.printDetails();
        	
        	System.out.println("\n");
            Car otherVersion  = ______________________ //<<3
            otherVersion.setName("Focus SE");
            otherVersion.setAutomaticClimateControl(true);
            otherVersion.setCruiseControl(true);
            otherVersion.printDetails();
           
 
        }
        catch (CloneNotSupportedException e)
        {
        	System.out.println("Object not cloneable");
            e.printStackTrace();
        }
    }
}